File: src/test/java/org/cerion/stocklist/overlays/ZigZagTest.java
package org.cerion.stocklist.overlays;

import org.cerion.stocklist.arrays.FloatArray;
import org.cerion.stocklist.functions.FunctionTestBase;
import org.cerion.stocklist.overlays.ZigZag;
import org.junit.Test;


public class ZigZagTest extends FunctionTestBase {

    @Test
    public void eval() throws Exception {

        int size = mPriceList.size();
        FloatArray arr = new ZigZag(20.0).eval(mPriceList);

        assertEqual(mPriceList.high(0), arr.get(0), "zigzag start");
        assertEqual(mPriceList.low(307), arr.get(307), "zigzag first low");
        assertEqual(mPriceList.high(349), arr.get(349), "zigzag first high");
        assertEqual(mPriceList.low(2957), arr.get(2957), "zigzag last low");
        assertEqual(mPriceList.high(3868), arr.get(3868), "zigzag last high");
        assertEqual(mPriceList.low(size-1), arr.get(size-1), "zigzag end");


    }

}

File: src/test/kotlin/overlays/ZigZagTest.kt
package org.cerion.stocklist.overlays

import org.cerion.stocklist.TestBase
import org.junit.Test


class ZigZagTest : TestBase() {

    @Test
    fun eval() {
        val arr = ZigZag(20.0).eval(priceList)

        assertEqual(priceList.high[0], arr[0], "zigzag start")
        assertEqual(priceList.low[307], arr[307], "zigzag first low")
        assertEqual(priceList.high[349], arr[349], "zigzag first high")
        assertEqual(priceList.low[2957], arr[2957], "zigzag last low")
        assertEqual(priceList.high[3868], arr[3868], "zigzag last high")
        assertEqual(priceList.low[size - 1], arr[size - 1], "zigzag end")
    }
}